pkg/ottl/ottlfuncs/func_time.go

```diff
@@ -0,0 +1,66 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/timeutils"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type TimeArguments[K any] struct {
+	Time     ottl.StringGetter[K] `ottlarg:"0"`
+	Format   ottl.StringGetter[K] `ottlarg:"1"`
```

Comment:
Let's change `Format` to be a `string` instead of a `StringGetter`. This brings Time inline with other functions that take a format, like a regex pattern, all of which force the value to be statically defined in the config via a string. Also, having a static string format will allow us to call `GetLocation` inside `time` before the return function.  This will give use the location to use so it doesn't need to be passed in.

Reply:
Changed!